capsules/extra/src/ieee802154/driver.rs

```diff
@@ -177,9 +177,41 @@ impl KeyDescriptor {
     }
 }
 
+/// Denotes the type of pending transmission. A `Parse(..)` PendingTX
+/// indicates that the 15.4 framer will need to form the packet header
+/// from the provided address and security level. A `Raw` PendingTX
+/// is formed by the userprocess and passes through the Framer unchanged.
+enum PendingTX {
+    Parse(u16, Option<(SecurityLevel, KeyId)>),
+    Raw,
+    Empty,
+}
+
+impl Default for PendingTX {
+    /// The default PendingTX is `Empty`
+    fn default() -> Self {
+        PendingTX::Empty
+    }
+}
+
+impl PendingTX {
+    /// Returns true if the PendingTX is not `Empty`
+    fn is_some(&self) -> bool {
+        match self {
+            PendingTX::Empty => false,
+            _ => true,
+        }
+    }
```

Comment:
I'm like 1% dubious about making our own `is_some()` method for something that isn't actually an Option. I don't feel strongly about it though, and maybe it's okay.

Reply:
I will update this to be `is_empty` (and will update the logic accordingly)